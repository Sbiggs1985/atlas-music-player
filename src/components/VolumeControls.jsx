import React, { useState } from "react";
import { Volume2 } from "lucide-react";

const VolumeControls = () => {
  const [volume, setVolume] = useState(50); // Default volume level at 50%

  const handleVolumeChange = (event) => {
    setVolume(event.target.value);
  };

  return (
    <div className="flex items-center gap-4">
      {/* Volume Icon */}
      <Volume2 className="w-6 h-6 text-black" />

      {/* Range Slider */}
      <input
        type="range"
        min="0"
        max="100"
        value={volume}
        onChange={handleVolumeChange}
        className="w-full h-2 bg-gray-200 rounded-lg appearance-none focus:outline-none focus:ring-2 focus:ring-black"
      />
    </div>
  );
};

export default VolumeControls;