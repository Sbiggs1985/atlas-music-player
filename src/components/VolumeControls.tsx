import React, { ChangeEvent } from "react";
import { Volume2 } from "lucide-react";

// Define props for external state control
interface VolumeControlsProps {
  volume: number;
  onChangeVolume: (value: number) => void;
}

const VolumeControls: React.FC<VolumeControlsProps> = ({ volume, onChangeVolume }) => {
  // Handle volume change event
  const handleVolumeChange = (event: ChangeEvent<HTMLInputElement>) => {
    onChangeVolume(Number(event.target.value));
  };

  return (
    <div className="flex items-center gap-4 w-full">
      {/* Volume Icon */}
      <Volume2 className="w-6 h-6 text-lightPurple dark:text-lightPurple" aria-hidden="true" />

      {/* Range Slider */}
      <input
        type="range"
        min="0"
        max="100"
        value={volume}
        onChange={handleVolumeChange}
        className="w-full h-2 bg-blackSlider dark:bg-blackSlider rounded-lg appearance-none focus:outline-none focus:ring-2 focus:ring-lightPurple"
        aria-label="Volume Control"
        aria-valuenow={volume}
      />

      {/* Volume Display */}
      <span className="text-sm text-lightPurple dark:text-lightPurple font-coolFont">
        {volume}%
      </span>
    </div>
  );
};

export default VolumeControls;

// TypeScript Integration
// - Added TypeScript typing for useState (number) and event handling (ChangeEvent<HTMLInputElement>).
// - handleVolumeChange Function: Updates the volume state dynamically as the slider moves.
// - Converts the slider value from string to number using Number().
