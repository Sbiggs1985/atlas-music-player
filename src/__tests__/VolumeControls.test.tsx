import { render, screen, fireEvent } from "@testing-library/react";
import VolumeControls from "@/components/VolumeControls";
import { describe, it, expect } from "vitest";

describe("VolumeControls Component", () => {
  it("renders the volume control component", () => {
    render(<VolumeControls />);
    expect(screen.getByLabelText("Volume Control")).toBeInTheDocument();
    expect(screen.getByText("50%")).toBeInTheDocument(); // Default volume
  });

  it("changes volume when slider is moved", () => {
    render(<VolumeControls />);
    
    const slider = screen.getByLabelText("Volume Control") as HTMLInputElement;
    fireEvent.change(slider, { target: { value: "80" } });

    expect(slider.value).toBe("80");
    expect(screen.getByText("80%")).toBeInTheDocument();
  });

  it("matches snapshot at default volume", () => {
    const { asFragment } = render(<VolumeControls />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot at max volume", () => {
    const { asFragment } = render(<VolumeControls />);
    
    const slider = screen.getByLabelText("Volume Control") as HTMLInputElement;
    fireEvent.change(slider, { target: { value: "100" } });

    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot at low volume", () => {
    const { asFragment } = render(<VolumeControls />);
    
    const slider = screen.getByLabelText("Volume Control") as HTMLInputElement;
    fireEvent.change(slider, { target: { value: "10" } });

    expect(asFragment()).toMatchSnapshot();
  });
});