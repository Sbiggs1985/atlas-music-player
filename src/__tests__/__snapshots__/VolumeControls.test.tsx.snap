// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls Component > matches snapshot at default volume 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-4 w-full"
  >
    <svg
      class="lucide lucide-volume2 w-6 h-6 text-lightPurple"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
     
    <input
      aria-label="Volume Control"
      class="w-full h-2 bg-blackSlider rounded-lg appearance-none focus:outline-none focus:ring-2 focus:ring-lightPurple"
      max="100"
      min="0"
      type="range"
      value="50"
    />
    <span
      class="text-sm text-lightPurple font-coolFont"
    >
      50%
    </span>
     
  </div>
</DocumentFragment>
`;

exports[`VolumeControls Component > matches snapshot at low volume 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-4 w-full"
  >
    <svg
      class="lucide lucide-volume2 w-6 h-6 text-lightPurple"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
     
    <input
      aria-label="Volume Control"
      class="w-full h-2 bg-blackSlider rounded-lg appearance-none focus:outline-none focus:ring-2 focus:ring-lightPurple"
      max="100"
      min="0"
      type="range"
      value="10"
    />
    <span
      class="text-sm text-lightPurple font-coolFont"
    >
      10%
    </span>
     
  </div>
</DocumentFragment>
`;

exports[`VolumeControls Component > matches snapshot at max volume 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-4 w-full"
  >
    <svg
      class="lucide lucide-volume2 w-6 h-6 text-lightPurple"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
     
    <input
      aria-label="Volume Control"
      class="w-full h-2 bg-blackSlider rounded-lg appearance-none focus:outline-none focus:ring-2 focus:ring-lightPurple"
      max="100"
      min="0"
      type="range"
      value="100"
    />
    <span
      class="text-sm text-lightPurple font-coolFont"
    >
      100%
    </span>
     
  </div>
</DocumentFragment>
`;
